parts/k8s/addons/secrets-store-csi-driver.yaml

```diff
@@ -365,8 +365,8 @@ spec:
             - name: mountpoint-dir
               mountPath: /var/lib/kubelet/pods
               mountPropagation: Bidirectional
-            - name: providers-dir
-              mountPath: /etc/kubernetes/secrets-store-csi-providers
```

Comment:
@aramase @chewong can you comment on the safety of this change?

Reply:
@jackfrancis Thank you for asking about the safety. That makes me think more about this and on second thought, mounting `/etc/kubernetes` also mounts all the Kubernetes certs into the pod. If the single driver pod is compromised it could result in exposing all those certs. Keeping in accordance with best practices, it might be better to mount the least possible content required to make this work which in this case would be `/etc/kubernetes/environment.json`. For the standalone deployments outside of aks-engine addons, we could provider an option to custom `volumeMounts` and `volumes` in the `secrets-store-csi-driver` helm charts that users can leverage to mount this custom environment file. WDYT?